### PR TITLE
Add exact field match to fix filtering on facets

### PR DESCRIPTION
**Pull request by markbreedlove**
Fixes https://issues.dp.la/issues/8398

Facilitate correct filtering based on clickthroughs from facets by doing an exact match on the facet values -- not a tokenized search.

See https://github.com/dpla/platform/pull/41

Also add some comments to the existing code.

### CODE REVIEW

**Comment by markbreedlove** [comment]
I've just rebased this on `develop` to pick up a bugfix there.

**Comment by anarchivist** [comment]
👍  :squash: ❓ 

Any other feedback, @AudreyAltman @no-reply @moltude?

File: lib/dpla/conditions.rb [hunk 1]
```diff
@@ -11,13 +11,19 @@ def to_s
     private
 
     def transform(conditions)
+      # When is `conditions' not an array or hash?  It seems that it should
```
Comment (no-reply) [inline lib/dpla/conditions.rb]: This looks right, but I think it's good practice to use exhaustive `case` statements. Falling back on `encode_uri(conditions)` seems pretty reasonable.

**Comment by no-reply** [comment]
:+1:

File: lib/dpla/conditions.rb [hunk 2]
```diff
@@ -11,13 +11,19 @@ def to_s
...
+      # always be one or the other, based on what I see elsewhere in the
+      # code. It can be an array of IDs when called by DPLA::Items.by_ids.
+      # It is usually a Hash.  --Mark B
```
Comment (AudreyAltman) [inline lib/dpla/conditions.rb]: I likewise could not find any case in which `conditions` was neither an Array nor a Hash.

**Comment by AudreyAltman** [comment]
:+1: from me.  Thanks, @markbreedlove!

File: lib/dpla/conditions.rb (hunk 1 above)
Comment (markbreedlove) [inline lib/dpla/conditions.rb]: I, too like the idea of an exhaustive case statement with good default behavior. The default case doesn't, at the moment, assert that the `conditions` getting passed to `encode_uri` is a String or has `#to_string`.  I'd prefer to see it raise an exception here, or assert that this is a string; but I think this is out of scope here, and I think the most I should do is modify my comment.

**Comment by markbreedlove** [comment]
I've just squashed this.
